src/java.desktop/share/classes/javax/swing/JLabel.java

```diff
@@ -1086,6 +1080,15 @@ public String getAccessibleName() {
             if (name == null) {
                 name = super.getAccessibleName();
             }
+
+
+            // If an icon is set but no text is set,
+            // the screen reader will say "image".
+            // cc jdk-8277497
+            if (((name == null) || name.isEmpty()) &&
+                    (JLabel.this.getIcon() != null)) {
+                name = ResourceBundle.getBundle("com.sun.accessibility.internal.resources.accessibility", Locale.getDefault()).getString("image");
+            }
```

Comment:
Probably it should somehow ask the icon itself about possible description? I guess the JLabel should work similar to Icon/ImageIcon/AccessibleImageIcon/etc when the text is empty but the icon is set. But I am not sure that the iicons are supported by the a11y in Swing, for example how the reader will cover the simple Icon? Will it say something?

Reply:
Done. thank you very much.

Reply:
I few questions to thinking about:
 - If the label and icon is not accessible then should we say something? Or we should ignore it like we do for any other non-accessible components?
 - Why the image text is used, don't we need to use the "javax.accessibility.AccessibleRole#ICON" role for such label/icon and allow the reader to say something standard for the icon. Does the voiceover has some default text for the icon/image when the alt text is not set?

Reply:
Thank you very much. We've made a change. A comparison of the behavior before and after the changes is below.

### On MAcOS.
#### State before our changes .:
- in a standalone label with an icon without text, VO says Image;
- in a standalone label with an icon with text VO speaks the text of the label;
- If a label without text with an icon is in the table VO says "Empty cell".

#### If you add just a role change:
- in a standalone label with an icon without text, VO says Image;
- in a standalone label with an icon with text VO speaks the text of the label;
- In the table for a label with an icon without text, when navigating horizontally, VO does not say anything;
- In the table for a label with an icon without text for vertical navigation, VO says "image cell".

Unacceptable option.

#### With a complete fix:
- in a standalone label with an icon without text, VO says Image;
- in a standalone label with an icon with text VO speaks the text of the label;
- In the table for a label with an icon without text, for any navigation, VO says "image".

### ON Windows.

#### State before our changes:
- in a standalone label with an icon and text, the label text is spoken;
- in a standalone label with an icon without text, the label is skipped;
- In the table, a label with an icon without text is not spoken, the column heading is spoken.

#### With full patch:
- in a standalone label with an icon and text, the label text is spoken;
- in a standalone label with an icon without text the "Image icon" is spoken;
- In the table, a label with an icon without text is pronounced as "Image".